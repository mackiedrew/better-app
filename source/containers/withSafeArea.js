import React, { PureComponent } from "react"
import { SafeAreaView } from "react-native"
import styled from "styled-components/native"

const SafeView = styled(SafeAreaView)`
  flex: 1;
`

export default Child =>
  class extends PureComponent {
    static propTypes = {}
    static defaultProps = {}
    render = () => (
      <SafeView>
        <Child {...this.props} />
      </SafeView>
    )
  }
