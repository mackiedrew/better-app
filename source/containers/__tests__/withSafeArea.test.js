import React from "react"
import { Text } from "react-native"

import renderer from "react-test-renderer"

import withSafeArea from "../withSafeArea"

const component = () => <Text>Test</Text>

describe("withSafeArea", () => {
  it("wraps component", () => {
    const WrappedComponent = withSafeArea(component)
    const render = renderer.create(<WrappedComponent />).toJSON()
    expect(render).toMatchSnapshot()
  })
})
