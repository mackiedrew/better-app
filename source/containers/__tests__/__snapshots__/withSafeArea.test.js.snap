// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withSafeArea wraps component 1`] = `
<RCTSafeAreaView
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Test
  </Text>
</RCTSafeAreaView>
`;
